### PR TITLE
Fix restart PlayCanvas Application issue.

### PR DESCRIPTION
**Pull request by H1Gdev**
**Test code**
https://github.com/H1Gdev/restart_playcanvas

1. new Application
2. start Application
3. wait 5[s]
4. destroy Application
5. new Application
6. start Application

**Expect**
-> cube entity color is the same as the first time.
**Actual**
-> cube entity color is black.

use pc.ModelHandler.DEFAULT_MATERIAL global variable directly, and keep instance at the first time.
(pc.ModelHandler.DEFAULT_MATERIAL.parameters.material_diffuse.scopeId etc.)

So use clone the following global default variables.

- pc.Scene.defaultMaterial
- pc.ModelHandler.DEFAULT_MATERIAL

### CODE REVIEW

**Comment by vkalpias** [comment]
I believe the point of the default material is to not be cloned for performances reasons.

**Comment by H1Gdev** [comment]
@vkalpias 
Thanks for your advice.

Reset pc.ModelHandler.DEFAULT_MATERIAL when destroy Application.

**Comment by H1Gdev** [comment]
@vkalpias 
I change commit.
ScopeSpace has latest ScopeId, so always get ScopeId from ScopeSpace.

**Comment by H1Gdev** [comment]
@guycalledfrank 
Please review this PR.

File: src/scene/forward-renderer.js [hunk 1]
```diff
@@ -2165,9 +2165,7 @@ pc.extend(pc, function () {
                         parameters = material.parameters;
                         for (paramName in parameters) {
                             parameter = parameters[paramName];
-                            if (!parameter.scopeId) {
-                                parameter.scopeId = device.scope.resolve(paramName);
-                            }
+                            parameter.scopeId = device.scope.resolve(paramName);
```
Comment (guycalledfrank) [inline src/scene/forward-renderer.js]: So why do we need to always call resolve for every uniform now?
I believe we should just reset the default material on init, not change anything in the render loop.

Comment (H1Gdev) [inline src/scene/forward-renderer.js]: @guycalledfrank 

Thanks for your review.

This issue is caused by old value remaining in global variable(pc.ModelHandler.DEFAULT_MATERIAL).
graphicsDevice ScopeSpace always has latest ScopeId.
So always get ScopeId reference from ScopeSpace is more safer.

**Comment by H1Gdev** [comment]
@guycalledfrank 

> I believe we should just reset the default material on init, not change anything in the render loop.

I fixed following commitment before.

> Reset pc.ModelHandler.DEFAULT_MATERIAL when destroy Application.

So, I revert.

File: src/framework/application.js [hunk 2]
```diff
@@ -1278,6 +1278,8 @@ pc.extend(pc, function () {
             // remove default particle texture
             pc.ParticleEmitter.DEFAULT_PARAM_TEXTURE = null;
 
+            pc.ModelHandler.resetDefaultMaterial();
```
Comment (willeastcott) [inline src/framework/application.js]: Would simply changing this line to pc.Scene.defaultMaterial.reset(); solve this issue? What do you think, @guycalledfrank?

Comment (H1Gdev) [reply src/framework/application.js]: @willeastcott 
Can not solve this issue.

This issue is caused by old value(instance) remaining in global variable(pc.ModelHandler.DEFAULT_MATERIAL.parameters[_name_].scopeId).

This field is excluded from StandardMaterial.reset() method.

**Comment by willeastcott** [comment]
Sorry, I didn't notice where the call to reset was being made. If I call pc.Scene.defaultMaterial.reset(); at the end of the pc.Application constructor, the example app you have works fine.

**Comment by H1Gdev** [comment]
@willeastcott 
Thanks for your confirmation.

This [playcanvas-latest.js](https://github.com/H1Gdev/restart_playcanvas/blob/master/playcanvas-latest.js) in example app is fixed this issue.

I added `pc.Scene.defaultMaterial.reset();` to original [playcanvas-latest.js](https://code.playcanvas.com/playcanvas-latest.js), but did not fix this issue.

**Comment by willeastcott** [comment]
I just tested the example app with the latest engine and the cube no longer turns black. So presumably this is fixed. I am closing this but if you believe there is still a problem, please re-open the PR.

**Pull request closed without merging**